### ISSUE TITLE
pytest --direct appears to be broken

### ISSUE THREAD

**Issue by timburke** [iotile/coretools#834]
See https://github.com/iotile/coretools/blob/master/iotiletest/iotile/fixture/device_fixture.py#L29

The `--direct` option is specified using `store_true` but it appears that the fixture expects it to contain a `connection_string`.  

My understanding is that we no longer use `--direct` anyway so we should deprecate and remove it.

@x24ling @ekkizogloy Do either of you use `--direct` from pytest for anything?  If not we should tag it for removal.

**Comment by x24ling** [comment]
I personally don't use it, but would it be useful for testing jlink connections?

**Comment by timburke** [comment]
I don't think it would be useful because we run rpcs over jlink as part of the tests and if rpcs works then the firmware is functional and you can connect via `uuid`.

**Comment by ekkizogloy** [comment]
I do not think i use that anywhere. It can be removed.
